File: Python/848.shifting-letters.py
'''
desc:尽可能把最大时间复杂度控制到O(n)
'''

from collections import Counter

class Solution:
	class Solution:
    def shiftingLetters(self, S: str, shifts: List[int]) -> str:
        curr = 0
        strs = ""
        for i in range(len(shifts), 0, -1):
            idx = i - 1
            curr += shifts[idx]
            curr = curr % 26
            c = S[idx]
            n = ord(c) + curr
            if n > 122:
                n = n -26
            strs = chr(n) + strs
        return strs

File: Go/848.shifting-letters.go

func shiftingLetters(S string, shifts []int) string {
    chars := []byte(S)
    curr := 0
    strs := ""
    for i := len(shifts)-1; i >= 0; i-- {
        curr += shifts[i]
        curr = curr % 26
        c := chars[i]
        n := int(c)
        n += curr
        if n > 122 {
            n -= 26
        }
        strs = string(rune(n)) + strs
    } 
    return strs
}
